lightning/restore/restore.go

```diff
@@ -403,15 +404,37 @@ func verifyCheckpoint(cfg *config.Config, taskCp *TaskCheckpoint) error {
 	return nil
 }
 
-func (rc *RestoreController) estimateChunkCountIntoMetrics() {
-	estimatedChunkCount := 0
+func (rc *RestoreController) estimateChunkCountIntoMetrics(ctx context.Context) error {
+	estimatedChunkCount := 0.0
 	for _, dbMeta := range rc.dbMetas {
 		for _, tableMeta := range dbMeta.Tables {
+			tableName := common.UniqueTable(dbMeta.Name, tableMeta.Name)
+			dbCp, err := rc.checkpointsDB.Get(ctx, tableName)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			fileChunks := make(map[string]float64)
+			for engineId, eCp := range dbCp.Engines {
+				if engineId == indexEngineID {
+					continue
+				}
+				for _, c := range eCp.Chunks {
+					if _, ok := fileChunks[c.Key.Path]; !ok {
+						fileChunks[c.Key.Path] = 0.0
+					}
+					remainChunkCnt := float64(c.Chunk.EndOffset-c.Chunk.Offset) / float64(c.Chunk.EndOffset-c.Key.Offset)
+					fileChunks[c.Key.Path] += remainChunkCnt
+				}
+			}
 			for _, fileMeta := range tableMeta.DataFiles {
+				if cnt, ok := fileChunks[fileMeta.FileMeta.Path]; ok {
+					estimatedChunkCount += cnt
+					continue
+				}
 				if fileMeta.FileMeta.Type == mydump.SourceTypeCSV {
 					cfg := rc.cfg.Mydumper
 					if fileMeta.Size > cfg.MaxRegionSize && cfg.StrictFormat && !cfg.CSV.Header {
-						estimatedChunkCount += int(fileMeta.Size / cfg.MaxRegionSize)
+						estimatedChunkCount += math.Round(float64(fileMeta.Size) / float64(cfg.MaxRegionSize))
```

Comment:
Does it match how we split files?

Reply:
yes, we split csv file by  `max-region-size` config. Each chunk maybe a little bigger than `max-region-size` since we always split chunk at the end of one line.